src/Storages/ObjectStorage/DataLakes/Iceberg/Mutations.cpp

```diff
@@ -94,13 +119,21 @@ std::optional<DeleteFileWriteResultWithStats> writeDataFiles(
     std::unordered_map<ChunkPartitioner::PartitionKey, std::unique_ptr<WriteBuffer>, ChunkPartitioner::PartitionKeyHasher> write_buffers;
     std::unordered_map<ChunkPartitioner::PartitionKey, OutputFormatPtr, ChunkPartitioner::PartitionKeyHasher> writers;
 
-    if (commands.front().type == MutationCommand::Type::DELETE)
+    DeleteFileWriteResultByPartitionKey update_data_result;
+    DeleteFileStatisticsByPartitionKey update_data_statistics;
+    std::unordered_map<ChunkPartitioner::PartitionKey, std::unique_ptr<WriteBuffer>, ChunkPartitioner::PartitionKeyHasher> update_data_write_buffers;
+    std::unordered_map<ChunkPartitioner::PartitionKey, OutputFormatPtr, ChunkPartitioner::PartitionKeyHasher> update_data_writers;
+
+    if (commands[0].type == MutationCommand::UPDATE || commands[0].type == MutationCommand::DELETE)
```

Comment:
Did not get it - why do we process `commands[0].type == MutationCommand::UPDATE` case  in this block along with delete?

Reply:
because `UPDATE` consists of "creating a delete position file" for the old data and "adding a new data file".
To get the old data blocks for the first part, we also handle updates with the same arguments but with the DELETE command type.